.pytool/Plugin/NestedPackageCheck/NestedPackageCheck.py

```diff
@@ -0,0 +1,60 @@
+# @file NestedPackageCheck.py
+#
+# A CI Check that verifies there are no nested packages in the package being tested.
```

Comment:
I suggest describing what "nested packages" means here or in the Readme since that comes up almost every time the topic does.

Reply:
I have updated the readme with more detailed information about what a nested package is, why it is bad, and the spec with the requirement that we are enforcing with this CI check.